### DISCUSSION TITLE
git post sync command

### DISCUSSION THREAD

**Discussion by lnnwvr** [argoproj/argo-cd#7172] [Q&A]
Hello experts,

i am looking for a way to add a git module and run automatically a git command after every sync. Is there anything build in esp. for the after sync step. I saw the hook annotation but i think this the wrong approach?

i.e. i want to call  git secret reveal or git-crypt unlock after sync to decrypt secrets automatically.

**Comment by jannfis** [comment]
I think you would need a [custom plugin](https://argo-cd.readthedocs.io/en/stable/operator-manual/custom_tools/) for this to work.

The plugin is given the Git checkout as source, and you can then run any further Git command on the local copy. Afterwards, you need to print the resulting manifests to stdout.
